examples/src/main/scala/com/cloudera/livy/examples/WordCountApp.scala

```diff
@@ -0,0 +1,243 @@
+      port: Int,
+      outputPath: String): ScalaJobHandle[Unit] = {
+    scalaClient.submit { context =>
+      context.createStreamingContext(15000)
+      val ssc = context.streamingctx
+      val sqlctx = context.sqlctx
+      val lines = ssc.socketTextStream(host, port, StorageLevel.MEMORY_AND_DISK_SER)
+      val words = lines.filter(filterEmptyContent(_)).flatMap(tokenize(_))
+      words.print()
+      words.foreachRDD { rdd =>
+        import sqlctx.implicits._
+        val df = rdd.toDF("word")
+        df.write.mode("append").json(outputPath)
+      }
+      ssc.start()
+      ssc.awaitTerminationOrTimeout(12000)
+      ssc.stop(false, true)
+    }
+  }
+
+  /**
+   * Submits a spark sql job to the livy server.
+   *
+   * The sql context job reads data frames from the given json path and executes
+   * a sql query to get the word with max count on the temp table created with data frames.
+   * @param inputPath Input path to the json data containing the words.
+   */
+  def getWordWithMostCount(inputPath: String): ScalaJobHandle[String] = {
+    scalaClient.submit { context =>
+      val sqlctx = context.sqlctx
+      try {
+        val rdd = sqlctx.read.json(inputPath)
+        rdd.registerTempTable("words")
+        val result = sqlctx.sql("select word, count(word) as word_count from words " +
+          "group by word order by word_count desc limit 1")
+        result.first().toString()
+      } catch {
+        case exception: IOException => {
+          throw new IOException("The input path does not have any data frame", exception)
+        }
+      }
+    }
+  }
+
+  private def filterEmptyContent(text: String): Boolean = {
+    text != null && !text.isEmpty
+  }
+
+  private def tokenize(text : String) : Array[String] = {
+    text.toLowerCase.replaceAll("[^a-zA-Z0-9\\s]", "").split("\\s+")
+  }
+
+  private def stopClient(): Unit = {
+    if (scalaClient != null) {
+      scalaClient.stop(true)
+      scalaClient = null;
+    }
+  }
+
+  /**
+   * Main method of the WordCount App. This method does the following
+   * - Validate the arguments.
+   * - Initializes the scala client of livy.
+   * - Uploads the required livy and app code jar files to the spark cluster needed during runtime.
+   * - Executes the streaming job that reads text-data from socket stream, tokenizes and saves
+   *   them as dataframes in JSON format in the given output path.
+   * - Executes the sql-context job which reads the data frames from the given output path and
+   * and returns the word with max count.
+   *
+   * @param args
+   *
+   * REQUIRED ARGUMENTS
+   * arg(0) - Livy server url.
+   * arg(1) - Output path to save the text read from the stream.
+   *
+   * OPTIONAL ARGUMENTS(Default values: arg(2) = 'localhost', arg(3) = 8086)
+   * arg(2) - Socket stream host that spark streaming will listen to.
+   * arg(3) - Socket stream port that spark streaming will listen to.
+   * Optional arguments need to be passed in a key=value way
```

Comment:
If they're passed as `key=value` then they don't need positional markers like `arg(2)`, right?

So just remove the previous lines and say "Remaining arguments are treated as `key=value` pairs. The following keys are recognized:"